### PR TITLE
Aspects

### PR DESCRIPTION
**Pull request by sils**
Continue to work on aspects! Glorious!

### CODE REVIEW

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2950/commits/f1168b5d17c02f9998406e8b85f4062bd8c1b7bf

File: coalib/bearlib/aspects/Metadata.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    )
+)
+
+Root.Metadata.new_subaspect(
+    "CommitMessage",
+    AspectDocumentation(
+        definition="""
+        Your commit message is important documentation associated with your
+        source code. It can help you to identify bugs (e.g. through
+        `git bisect`) or find missing information about unknown source code
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: `finding more information about the original author` for `git blame`?

File: coalib/bearlib/aspects/Metadata.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        """
+    )
+)
+
+Root.Metadata.CommitMessage.new_subaspect(
+    "Shortlog",
+    AspectDocumentation(
+        definition="""
+        Your commit shortlog is the first line of your commit message. It is
+        the most crucial part of your commit message and summarizes the change
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: drop the `of your commit message` the second time - repetitive

Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: also `summarizes the change**s**`

File: coalib/bearlib/aspects/Metadata.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        (True, False),
+        True
+    ),)
+)
+
+Root.Metadata.CommitMessage.Shortlog.new_subaspect(
+    "TrailingPeriod",
+    AspectDocumentation(
+        definition="""
+        Some projects force not to use - or use - trailing periods in the commit
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: this sounds a bit cryptic - phrase it `Some projects force to use - or not use - ...`? I know that the default value is `False`, but this makes more sense IMO. We shouldn't need to change the description based on the default value.

File: coalib/bearlib/aspects/Metadata.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        message shortlog (first line). (Not having periods is more common.)
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: should this really be in the description of the aspect?

File: coalib/bearlib/aspects/Metadata.py [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        Add coverage pragma
+        Compatability: Add coverage pragma
+        add coverage pragma
+        Compatability: add coverage pragma
+        """,
+        example_language="English",
+        importance_reason="Consistent commit messages are easier to read "
+                          "through.",
+        fix_suggestions="""
+        Upper or lower the first character. If your message starts with an
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: upper or lower **case** the first char

File: coalib/bearlib/aspects/__init__.py [hunk 6]
```diff
@@ -294,6 +294,9 @@ def __getattr__(self, subaspect):
         """
         if subaspect in self.subaspects:
             return self.subaspects[subaspect]
+
+        if subaspect == '__wrapped__':  # pragma: no cover
+            return None  # Needed for doctest
```
Comment (adtac) [inline coalib/bearlib/aspects/__init__.py]: should this change be in this commit?

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2950/commits/ff41b96986d8c00c6bec7dcc005809a6879ebbcc https://github.com/coala/coala/pull/2950/commits/2016279f5d1e49894c7843d51e0479553e661598

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2950/commits/d7093dc73618a17aeca8cf0f6d3f79ae9e35e66f https://github.com/coala/coala/pull/2950/commits/c84e2f7ec96ffe0bf971e4d75293417ebd7c4225

File: coalib/bearlib/aspects/Metadata.py [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+        (e.g. FIX, ...) or others. Some projects prefer not using colons
+        specifically: consistency is key.
+        """,
+        fix_suggestions="""
+        Add or remove the colon according to the commit message guidelines.
+        """
+    ),
+    settings=(AspectSetting(
+        "shortlog_colon",
+        "Wether or not the shortlog has to contain a colon.",
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: `Whether` typo

File: coalib/bearlib/aspects/Metadata.py [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+                          "through.",
+        fix_suggestions="""
+        Convert your first character to upper/lower case. If your message starts
+        with an identifier, consider rephrasing. Usually starting with a verb is
+        a good idea.
+        """
+    ),
+    settings=(AspectSetting(
+        "shortlog_starts_upper_case",
+        "Wether or not the shortlog (first line) of a commit message should "
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: Ctrl + F `Wether`

File: coalib/bearlib/aspects/Metadata.py [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+    ),)
+)
+
+Root.Metadata.CommitMessage.Shortlog.new_subaspect(
+    "Tense",
+    AspectDocumentation(
+        definition="""
+        Most projects have a convention on which tense to use in the commit
+        shortlog (the first line of the commit message). Usually the imperative
+        tense is chosen for its brevity.
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: this sentence shouldn't be in the aspect description

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2950/commits/58dc991cdec1b9db56fce5f7999eb683012da51b

**Comment by sils** [comment]
a4ce520 Add Metadata Aspects
7856ed4 AspectDocumentation: Remove `cause` member
738c618 aspects: Add Root aspect
93c8e57 aspects: Remove unused import
be1ed7b aspects: Add NameError message
2949487 aspects: Use immutable default values
reack

**Comment by sils** [comment]
@rultor merge

**Pull request merged**